v2_dev build not parsing JSX in javascript

### What version of Remix are you using?

1.19.1

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

see https://github.com/brentmjohnson/remixQuickStart for reproducible example.

- init new project from template `npx create-remix@latest --template https://github.com/remix-run/remix/tree/main/templates/express ./ --no-typescript`

  notice tsx files are still created instead of js

- replace the following files with valid js versions:
  ```
  app/routes/_index.tsx
  app/entry.client.tsx
  app/entry.server.tsx
  app/root.tsx
  ```

- npm run build succeeds:
  ```
  $ npm run build

  > build
  > remix build

   info  building... (NODE_ENV=production)
   info  built (240ms)
  ```

- npm run dev fails:
  ```
  $ npm run dev
  
  > dev
  > remix dev --manual -c "node server.js"
  
  
   💿  remix dev
  
   info  building...
  ✘ [ERROR] Expected ">" but found "/"
  
      app/entry.client.js:23:20:
        23 │       <RemixBrowser />
           │                     ^
           ╵                     >
  
  
  ✘ [ERROR] Expected ">" but found "lang"
  
      app/root.js:55:15:
        55 │   return <html lang="en">
           │                ~~~~
           ╵                >
  
  
  ✘ [ERROR] Expected ">" but found "style"
  
      app/routes/_index.js:30:14:
        30 │   return <div style={{
           │               ~~~~~
           ╵               >
  ```

### Expected Behavior

remix dev build occurring in v2_dev should succeed like remix production build

### Actual Behavior

remix dev build occurring in v2_dev fails

--- a/.changeset/lovely-drinks-bow.md
+++ b/.changeset/lovely-drinks-bow.md
@@ -1,0 +1,5 @@
+---
+"@remix-run/dev": patch
+---
+
+allow jsx in js files during hmr

--- a/packages/remix-dev/compiler/js/plugins/hmr.ts
+++ b/packages/remix-dev/compiler/js/plugins/hmr.ts
@@ -139,7 +139,7 @@ declare global {
             sourceCode,
             output: {
               contents: resultCode,
-              loader: args.path.endsWith("x") ? "tsx" : "ts",
+              loader: "tsx",
               resolveDir: path.dirname(args.path),
             },
           };

